pkg/controller/istiocontrolplane/inputs.go

```diff
@@ -31,12 +62,11 @@ func (i *IstioRenderingInput) GetTargetNamespace() string {
 }
 
 // GetProcessingOrder returns the order in which the rendered charts should be processed.
-func (i *IstioRenderingInput) GetProcessingOrder(manifests helmreconciler.ChartManifestsMap) ([]string, error) {
-	ordering := make([]string, 0, len(manifests))
+func (i *IstioRenderingInput) GetProcessingOrder(_ helmreconciler.ChartManifestsMap) (helmreconciler.ComponentNameToListMap, helmreconciler.DependencyWaitCh) {
+	return componentDependencies, dependencyWaitCh
+}
 
-	// everything else to the end
-	for chart := range manifests {
-		ordering = append(ordering, chart)
-	}
-	return ordering, nil
+func buildInstallTree() {
+	// Starting with root, recursively insert each first level child into each node.
+	helmreconciler.InsertChildrenRecursive(name.IstioBaseComponentName, installTree, componentDependencies)
```

Comment:
recursion is taught by universities to make programming harder and weed out students...  It would be better if this could be unrolled ;)